### PR TITLE
Doc processing cellstatistics

### LINKED ISSUE #5793: [feature] native cellstatistics algorithm
Original commit: https://github.com/qgis/QGIS/commit/bc74bbf5bddf1834326a6976df94be407767cbb3 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by root676**
**Description:**
This PR adds documentation on the **Cell statistics** processing algorithm that becomes available in 3.16. I hope the basic structure of the alg documentation is in line with the documentation guidelines. 

I am looking forward to any feedback/reviews as this is my first PR to the documentation! 
Fixed #5793

### CODE REVIEW

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 1]
```diff
@@ -7,6 +7,157 @@ Raster analysis
...
+- ``Median``: If the number of input layers is even, the median will be calculated as the arithmetic mean of the two middle values of the ordered cell input values.
+
+- ``Minority/Majority``: If no unique minority or majority could be found, the result is NoData, except all input cell values are equal.
+
+
+.. figure:: img/cellstatistics_mean.png
+  :align: center
+
+  Example for the cell statistics operation ``mean`` on three input rasters,
+  taking NoData cells (grey) into account. Raster data is based on `ESRI Cell statistics example. <https://pro.arcgis.com/en/pro-app/tool-reference/spatial-analyst/how-cell-statistics-works.htm>`_
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:50]: It eould be nice if we could avoid citing other, competing GIS software...

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 2]
```diff
@@ -7,6 +7,157 @@ Raster analysis
...
+
+- ``Median``: If the number of input layers is even, the median will be calculated as the arithmetic mean of the two middle values of the ordered cell input values.
+
+- ``Minority/Majority``: If no unique minority or majority could be found, the result is NoData, except all input cell values are equal.
+
+
+.. figure:: img/cellstatistics_mean.png
+  :align: center
+
+  Example for the cell statistics operation ``mean`` on three input rasters,
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:49]:
```suggestion
  Example for the cell statistics operation ``mean`` on two input rasters,
```
Ican only see two input rasters, but the values in the output raster indicates that on input raster is missing.

**Review by havatv — COMMENTED** [review]
Thanks, @root676!
 structure is fine, and this looks good. I have provided a couple of commments separately, that I hope you will consider.
Also, outputs are normally "returned" / listed in alphabetical order in the log window, so I suggest that we also sort them in  the documentation.

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 3]
```diff
@@ -7,6 +7,157 @@ Raster analysis
...
+     - Type
+     - Description
+   * - **Input layers**
+     - ``INPUT``
+     - [multipleLayers]
+     - Input raster layers
+   * - **Statistic**
+     - ``STATISTIC``
+     - [enumeration]
+       Default: 0
```
Comment (DelazJ) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:72]:
```suggestion

       Default: 0
```

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 4]
```diff
@@ -7,6 +7,157 @@ Raster analysis
       :local:
       :depth: 1
 
+.. _qgiscellstatistics:
+
+Cell statistics |316|
+---------------------
+
+The Cell statistics algorithm computes a value for each
```
Comment (DelazJ) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:15]:
```suggestion
Combines multiple raster layers into one and computes a value for each 
```
- no "this algorithm..." at the beginning
- and the alg does not just output stats; it also creates the raster, right?

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 5]
```diff
@@ -7,6 +7,157 @@ Raster analysis
...
+data type present in the input datasets except when using the 
+functions ``Mean`` and ``Standard deviation`` (data type is always ``Float32``) 
+or ``Count`` and ``Variety`` (data type is always ``Int32``).
+
+- ``Count``: The count statistic will always result in the number of cells without NoData values at the current cell location.
+
+- ``Median``: If the number of input layers is even, the median will be calculated as the arithmetic mean of the two middle values of the ordered cell input values.
+
+- ``Minority/Majority``: If no unique minority or majority could be found, the result is NoData, except all input cell values are equal.
+
```
Comment (DelazJ) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:44]:
```suggestion
- ``Count``: The count statistic will always result in the number of cells
  without NoData values at the current cell location.
- ``Median``: If the number of input layers is even, the median will be calculated
  as the arithmetic mean of the two middle values of the ordered cell input values.
- ``Minority/Majority``: If no unique minority or majority could be found,
  the result is NoData, except all input cell values are equal.
```
shorten lines

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 2 above)
Comment (DelazJ) [reply docs/user_manual/processing_algs/qgis/rasteranalysis.rst:49]: Everything looks good to me here: I see three inputs and a mean result layer

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 1 above)
Comment (DelazJ) [reply docs/user_manual/processing_algs/qgis/rasteranalysis.rst:50]: @root676 maybe could we use another stat and rasters?

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 6]
```diff
@@ -7,6 +7,157 @@ Raster analysis
...
+output value is defined as a function of all overlaid 
+cell values of the input rasters.
+
+By default, a NoData cell in ANY of the input layers will result in a
+NoData cell in the output raster.
+If the :guilabel:`Ignore NoData values` option is checked,
+then NoData inputs will be ignored in the statistic calculation. This
+may result in NoData output for locations where all cells are NoData.
+
+The :guilabel:`reference layer` parameter specifies an existing raster layer to
```
Comment (DelazJ) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:26]:
```suggestion
The :guilabel:`Reference layer` parameter specifies an existing raster layer to
```

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 1 above)
Comment (root676) [reply docs/user_manual/processing_algs/qgis/rasteranalysis.rst:50]: @DelazJ you are right, I'll rework the example.

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 2 above)
Comment (root676) [reply docs/user_manual/processing_algs/qgis/rasteranalysis.rst:49]: @havatv is there an issue with the PNG file? It displays 3 input rasters.

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 7]
```diff
@@ -7,6 +7,160 @@ Raster analysis
...
+    - ``EXTENT``
+    - [extent]
+    - The spatial extent of the output raster layer
+  * - **Height in pixels**
+    - ``HEIGHT_IN_PIXELS``
+    - [integer]
+    - The height in pixels of the output raster layer
+  * - **Output raster**
+    - ``OUTPUT``
+    - [same as input]
```
Comment (DelazJ) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:143]:
```suggestion
    - [raster]
```
"same as input" is used if the output may vary depending on the input. Eg if you can enter line, or polygon and in output you get line or polygon respectively. Here, there's no doubt you get raster as output.

**Review by DelazJ — APPROVED** [review]
One last suggestion. And LGTM.
I like the new screenshot, much much better ( and different from our brother's)

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 2 above)
Comment (havatv) [reply docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: > @havatv is there an issue with the PNG file? It displays 3 input rasters.

Today I see three input rasters when I use "View" (not so the last time- strange...)

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 8]
```diff
@@ -7,6 +7,160 @@ Raster analysis
...
+  calculated as the arithmetic mean of the two middle values of the ordered cell input values.
+
+- ``Minority/Majority``: If no unique minority or majority could be found,
+  the result is NoData, except all input cell values are equal.
+
+
+.. figure:: img/cell_statistics_all_stats.png
+  :align: center
+
+  Example for all statistic functions while taking NoData cells (grey) into account.
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:53]:
```suggestion
  Example with all the statistic functions. ``NoData`` cells (grey) are taken into account.
```

**Comment by root676** [comment]
@havatv @DelazJ thank you very much for the detailed review and feedback on my first contribution to the QGIS documentation! I just pushed the last changes you requested.

File: docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 9]
```diff
@@ -7,6 +7,160 @@ Raster analysis
...
+  calculated as the arithmetic mean of the two middle values of the ordered cell input values.
+
+- ``Minority/Majority``: If no unique minority or majority could be found,
+  the result is NoData, except all input cell values are equal.
+
+
+.. figure:: img/cell_statistics_all_stats.png
+  :align: center
+
+  Example with all the statistic functions.``NoData`` cells (grey) are taken into account.
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/rasteranalysis.rst:53]:
```suggestion
  Example with all the statistic functions. ``NoData`` cells (grey) are taken into account.
```

**Comment by havatv** [comment]
Thanks again, @root676 !

**Pull request merged**